src/v/config/rest_authn_endpoint.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/rest_authn_endpoint.h"
+
+#include "kafka/client/exceptions.h"
+#include "model/metadata.h"
+#include "utils/string_switch.h"
+
+namespace config {
+
+constexpr inline std::string_view to_string_view(rest_authn_type m) {
+    switch (m) {
+    case rest_authn_type::none:
+        return "none";
+    case rest_authn_type::http_basic:
+        return "http_basic";
+    default:
+        return "invalid";
+    }
+}
+
+template<>
+std::optional<rest_authn_type>
+from_string_view<rest_authn_type>(std::string_view sv) {
+    return string_switch<std::optional<rest_authn_type>>(sv)
+      .match("none", rest_authn_type::none)
+      .match("http_basic", rest_authn_type::http_basic)
+      .default_match(std::nullopt);
+}
+
+std::ostream& operator<<(std::ostream& os, const rest_authn_endpoint& ep) {
+    fmt::print(os, "{{{}:{}:{}}}", ep.name, ep.address, ep.authn_type);
+    return os;
+}
+
+} // namespace config
+
+namespace YAML {
+
+Node convert<config::rest_authn_endpoint>::encode(const type& rhs) {
+    Node node;
+    node["name"] = rhs.name;
+    node["address"] = rhs.address.host();
+    node["port"] = rhs.address.port();
+    if (rhs.authn_type) {
+        node["authentication_method"]["authentication_type"] = ss::sstring(
+          to_string_view(*rhs.authn_type));
+    }
+    return node;
+}
```

Comment:
Continuing the discussion from #6338 here. Original thread is [here](https://github.com/redpanda-data/redpanda/pull/6338#discussion_r971259006)


>  @NyaliaLui maybe adding example to cover letter can help surface the discussion?

Added in.

I'm in favor of using a single string to represent auth types. This project is not intended to cover the various ways we can represent multiple authentication methods.

```yaml
authentication_method: none, mtls, http_basic
```

I think this commit is bringing to light that we need to discuss and redesign how to represent multiple authentication methods in Redpanda. That discussion, however, could take some time and would end up blocking this PR and basic auth support. I believe we already have a project around this subject? I think it's called "central auth config" or something like that.

Reply:
Yeah, it might be best to follow whats done on the Kafka API and have:
```yaml
pandaproxy_api:
  authentication_method: http_basic
```

Reply:
I sitll need to make this change. It has cascading effects so I left it last.